packages/renderer/src/lib/image/ImageDetailsCheck.svelte

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+import type { ImageCheck } from '@podman-desktop/api';
+import type { ImageCheckerInfo } from '../../../../main/src/plugin/api/image-checker-info';
+import type { ImageInfoUI } from './ImageInfoUI';
+import Markdown from '../markdown/Markdown.svelte';
+import Fa from 'svelte-fa';
+import { faCheckCircle, faExclamationCircle, faStethoscope } from '@fortawesome/free-solid-svg-icons';
+import Button from '../ui/Button.svelte';
+import WarningMessage from '../ui/WarningMessage.svelte';
+import type { Unsubscriber } from 'svelte/store';
+import { onDestroy, onMount } from 'svelte';
+import { imageCheckerProviders } from '/@/stores/image-checker-providers';
+
+export let image: ImageInfoUI;
+
+interface CheckUI {
+  provider: ImageCheckerInfo;
+  check: ImageCheck;
+}
+
+let providers: ImageCheckerInfo[];
+let results: CheckUI[] = [];
+let cancellableTokenId: number = 0;
+
+let remainingProviders: number;
+let aborted = false;
+
+let providersUnsubscribe: Unsubscriber;
+
+onMount(async () => {
+  providersUnsubscribe = imageCheckerProviders.subscribe(providers => {
+    callProviders(providers);
+  });
+});
+
+onDestroy(() => {
+  // unsubscribe from the store
+  if (providersUnsubscribe) {
+    providersUnsubscribe();
+  }
+});
+
+async function callProviders(_providers: readonly ImageCheckerInfo[]) {
+  providers = [..._providers];
+  cancellableTokenId = await window.getCancellableTokenSource();
+  remainingProviders = providers.length;
+
+  _providers.forEach(provider => {
+    window
+      .imageCheck(provider.id, image.name, cancellableTokenId)
+      .then(_result => {
+        remainingProviders--;
+        _result?.checks.forEach(check => {
+          results.push({
+            provider,
+            check,
+          });
+        });
+        results.sort((a, b) => {
+          const orderStatus = ['failed', 'success'];
+          const orderSeverity = ['critical', 'high', 'medium', 'low', undefined];
+          const orderProvider = providers.map(p => p.id).sort();
+
+          const statusA = orderStatus.indexOf(a.check.status);
+          const statusB = orderStatus.indexOf(b.check.status);
+          if (statusA !== statusB) {
+            return statusA - statusB;
+          }
+
+          const severityA = orderSeverity.indexOf(a.check.severity);
+          const severityB = orderSeverity.indexOf(b.check.severity);
+          if (severityA !== severityB) {
+            return severityA - severityB;
+          }
+
+          const providerA = orderProvider.indexOf(a.provider.id);
+          const providerB = orderProvider.indexOf(b.provider.id);
+          if (providerA !== providerB) {
+            return providerA - providerB;
+          }
+          return 0;
+        });
+        results = [...results];
+      })
+      .catch((e: unknown) => console.log('check error', e));
```

Comment:
it looks like user is not notified that there was an error there
results should contain the error no ?